pkgs/os-specific/linux/kernel/xanmod-kernels.nix

```diff
@@ -29,9 +29,9 @@ let
       # AMD P-state driver
       X86_AMD_PSTATE = lib.mkOverride 60 yes;
 
-      # Google's BBRv2 TCP congestion Control
-      TCP_CONG_BBR2 = yes;
-      DEFAULT_BBR2 = yes;
+      # Google's BBRv3 TCP congestion Control
```

Comment:
Why do we even use `structuredExtraConfig`? Isn't https://github.com/xanmod/linux/blob/6.4/CONFIGS/xanmod/gcc/config_x86-64-v1 being used? Or am I just clueless about how kernel build configs work?

Reply:
Every kernel follows https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/common-config.nix.

Reply:
Does that not mean that there are potentially things these custom kernels offer but they aren't enabled in nixpkgs?

Reply:
Yes, that was the main motivation behind https://github.com/NixOS/nixpkgs/pull/128007.

Reply:
I see, but this still requires lots of manual work and it's very easy to miss something. We're probably missing something from xanmod already.

Reply:
@xanmod, could you announce when configuration options change so that downstrams such as us can adjust?

Reply:
A list of non-mainline configuration options would also be great.

Reply:
> @xanmod, could you announce when configuration options change so that downstrams such as us can adjust?

https://github.com/xanmod/linux/commit/HEAD
https://github.com/xanmod/linux/releases

> A list of non-mainline configuration options would also be great.

Diff the distro's stock kernel config to see the differences.

Reply:
`pkgs.linux_xanmod_latest.configfile` and https://github.com/xanmod/linux/blob/6.4/CONFIGS/xanmod/gcc/config_x86-64-v1 seem pretty different when diffing, @Atemu @fortuneteller2k wanna take a look at it?

Reply:
Sorry for the late reply but yes, I will take a look this weekend @Kranzes.

@xanmod could you tell us whether there is some base config that you build upon so that we can gather the set of xanmod-specific config flags? Diffing our config against xanmod contains a lot of noise from things we just do differently in the regular kernel options.